packages/core/saved-objects/core-saved-objects-migration-server-internal/src/core/build_active_mappings.ts

```diff
@@ -159,6 +159,9 @@ export function getBaseMappings(): IndexMapping {
       coreMigrationVersion: {
         type: 'keyword',
       },
+      typeMigrationVersion: {
```

Comment:
can't comment at the right place, but we should remove the mapping for `migrationVersion` on line 124

Reply:
We cannot remove that yet as we are updating mappings when there are still documents with the old `migrationVersion` properties. That happens before we apply core migrations.

Reply:
Removing a field will mean the migration algorithm doesn't think we've made an incompatible mappings change, so it would re-use the index instead of creating a new one. But mappings updates are commutative, so even if we update the mappings without `migrationVersion` ES will still keep it on existing indices.

Whenever we create a new index we should run the OUTDATED_DOCUMENTS_TRANSFORM* steps before we apply the final mappings to the index.

So I can't really imagine where this would fail?

Reply:
So I think this is because we don't check for coreMigrationVersion in the outdatedDocumentsQuery. So unless:
a) there's incompatible mappings and we reindex all documents
b) all saved object types have a new typeMigrationVersion (unlikely)

Some documents will be the old coreMigrationVersion and require the migrationVersion mapping.